From e2d0f0265cd28d397f86848dc860e45107e08ead Mon Sep 17 00:00:00 2001
From: "Ivan T. Ivanov" <iivanov@suse.de>
Date: Wed, 18 Jun 2025 10:47:43 +0300
Subject: [PATCH] ARM: dts: bcm2712: Remove DMA support

No upstream "brcm,bcm2712-dma" DMA driver support.
Drivers using these DMA channels will work even without them.
---
 .../arm/boot/dts/broadcom/bcm2711-rpi-ds.dtsi |  4 ----
 .../boot/dts/overlays/bcm2712d0-overlay.dts   | 21 -------------------
 arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi | 15 -------------
 .../boot/dts/broadcom/bcm2712d0-rpi-5-b.dts   | 11 ----------
 4 files changed, 51 deletions(-)

diff --git a/arch/arm/boot/dts/broadcom/bcm2711-rpi-ds.dtsi b/arch/arm/boot/dts/broadcom/bcm2711-rpi-ds.dtsi
index fcf70190737..0c4c90a292a 100644
--- a/arch/arm/boot/dts/broadcom/bcm2711-rpi-ds.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm2711-rpi-ds.dtsi
@@ -17,10 +17,6 @@ __overrides__ {
 		sd = <&emmc2>,"status";
 
 		sd_poll_once = <&emmc2>, "non-removable?";
-		spi_dma4 = <&spi0>, "dmas:0=", <&dma40>,
-			   <&spi0>, "dmas:8=", <&dma40>;
-		i2s_dma4 = <&i2s>, "dmas:0=", <&dma40>,
-			   <&i2s>, "dmas:8=", <&dma40>;
 	};
 
 	scb: scb {
diff --git a/arch/arm/boot/dts/overlays/bcm2712d0-overlay.dts b/arch/arm/boot/dts/overlays/bcm2712d0-overlay.dts
index 1ee74234e80..d7a4131f5ff 100644
--- a/arch/arm/boot/dts/overlays/bcm2712d0-overlay.dts
+++ b/arch/arm/boot/dts/overlays/bcm2712d0-overlay.dts
@@ -44,25 +44,4 @@ __overlay__ {
 			interrupts = <GIC_SPI 120 IRQ_TYPE_LEVEL_HIGH>;
 		};
 	};
-
-	fragment@5 {
-		target = <&spi10>;
-		__overlay__ {
-			dmas = <&dma40 3>, <&dma40 4>;
-		};
-	};
-
-	fragment@6 {
-		target = <&hdmi0>;
-		__overlay__ {
-			dmas = <&dma40 (12|(1<<30)|(1<<24)|(10<<16)|(15<<20))>;
-		};
-	};
-
-	fragment@7 {
-		target = <&hdmi1>;
-		__overlay__ {
-			dmas = <&dma40 (13|(1<<30)|(1<<24)|(10<<16)|(15<<20))>;
-		};
-	};
 };
diff --git a/arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi b/arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi
index 4a1a59388d8..a07288f88b3 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi
@@ -249,21 +249,6 @@ &dma40 {
 	brcm,dma-channel-mask = <0x07c0>;
 };
 
-&hdmi0 {
-	dmas = <&dma40 (10|(1<<30)|(1<<24)|(10<<16)|(15<<20))>;
-	dma-names = "audio-rx";
-};
-
-&hdmi1 {
-	dmas = <&dma40 (17|(1<<30)|(1<<24)|(10<<16)|(15<<20))>;
-	dma-names = "audio-rx";
-};
-
-&spi10 {
-	dmas = <&dma40 6>, <&dma40 7>;
-	dma-names = "tx", "rx";
-};
-
 &usb {
 	power-domains = <&power RPI_POWER_DOMAIN_USB>;
 };
diff --git a/arch/arm64/boot/dts/broadcom/bcm2712d0-rpi-5-b.dts b/arch/arm64/boot/dts/broadcom/bcm2712d0-rpi-5-b.dts
index d06536bc759..dbcbce07593 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2712d0-rpi-5-b.dts
+++ b/arch/arm64/boot/dts/broadcom/bcm2712d0-rpi-5-b.dts
@@ -94,14 +94,3 @@ &uart10 {
 	interrupts = <GIC_SPI 120 IRQ_TYPE_LEVEL_HIGH>;
 };
 
-&spi10 {
-	dmas = <&dma40 3>, <&dma40 4>;
-};
-
-&hdmi0 {
-	dmas = <&dma40 (12|(1<<30)|(1<<24)|(10<<16)|(15<<20))>;
-};
-
-&hdmi1 {
-	dmas = <&dma40 (13|(1<<30)|(1<<24)|(10<<16)|(15<<20))>;
-};
-- 
2.43.0

